sqlmesh/core/model/kind.py

```diff
@@ -152,6 +152,26 @@ def __repr__(self) -> str:
         return str(self)
 
 
+class OnSchemaChange(str, Enum):
```

Comment:
Although I like the simplicity of `OnSchemaChange`, it is a bit misleading. It really is `OnDestructiveSchemaChange`. Also I don't currently intuitively link `OnSchemaChange` with `allow_destructive_models`. So I think this naming still needs some work. 

What about `OnDestructiveChange`? Or I'm thinking we want to find something better than "Destructive"?

Reply:
Agreed - so I'll change both the enum name here and the kind option (`on_schema_change` --> `on_destructive_change`)?